code/xcvm/cosmwasm/contracts/gateway/src/interpreter.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+	contract::INSTANTIATE_INTERPRETER_REPLY_ID,
+	error::{ContractError, Result},
+	events::make_event,
+	network::load_this,
+	state::{self, interpreter},
+};
+use cosmwasm_std::{
+	to_binary, Deps, DepsMut, Reply, Response, StdError, StdResult, Storage, SubMsg, WasmMsg,
+};
+use cw_xc_interpreter::contract::{
+	XCVM_INTERPRETER_EVENT_DATA_ORIGIN, XCVM_INTERPRETER_EVENT_PREFIX,
+};
+use xc_core::{CallOrigin, InterpreterOrigin, NetworkId};
+
+use crate::{auth, prelude::*};
+
+pub(crate) fn force_instantiate(
+	auth: auth::Admin,
+	gateway: Addr,
+	deps: DepsMut,
+	user_origin: Addr,
+) -> Result {
+	let config = load_this(deps.storage)?;
+	let interpreter_code_id = match config.gateway.expect("expected setup") {
+		GatewayId::CosmWasm { interpreter_code_id, .. } => interpreter_code_id,
+	};
+
+	let call_origin = CallOrigin::Local { user: user_origin };
+	let interpreter_origin = InterpreterOrigin {
+		user_origin: call_origin.user(config.network_id),
+		salt: b"default".to_vec(),
```

Comment:
Should salt be argument instead? (And if we don’t care about salt here maybe just set it to empty value?)